frontend/src/components/MainComponent/MainComponent.tsx

```diff
@@ -283,6 +299,16 @@ function MainComponent({
 		setMessages((messages: ChatMessage[]) => [resetMessage, ...messages]);
 	}
 
+	function setChatModelId(modelId: string) {
+		if (!chatModel) {
+			console.error(
+				'You are trying to change the id of the chatModel but it has not been loaded yet'
```

Comment:
Not convinced this error message will help a future developer work out what's going on here. Can you come up with something indicating we only expect chat model to be changed on Sandbox level?

Reply:
"available models" arrives under the startResponse, along with chatModel. So if the game is loaded up or refreshed on sandbox, then the available models and chatModel arrive at the same time. However if the user switches from any level to sandbox, then there will be some amount of time (usually negligible) where the chat model dropdown and choose button are rendered, but the chatModel is not yet loaded.

Reply:
Okay, conditionally rendering the select model bits depending on chatModel being loaded. Wondering if we're doing this then we should also not render the model configuration sliders...
![image](https://github.com/ScottLogic/prompt-injection/assets/118171430/b19a5f3c-1ef9-4529-8fa3-a9be9a886d02)

Reply:
> "available models" arrives under the startResponse, along with chatModel. So if the game is loaded up or refreshed on sandbox, then the available models and chatModel arrive at the same time. However if the user switches from any level to sandbox, then there will be some amount of time (usually negligible) where the chat model dropdown and choose button are rendered, but the chatModel is not yet loaded.

Right, that makes sense now 😅 

I'd say if we're going down this route, then we should conditionally render the sliders as you suggest. Can you see how that looks - in the code I mean, check it's not too gnarly?

Reply:
As discussed in the standup, we'll leave this for a future issue, as it has some (re)design implications.